lib/ReportHistoryStore.jsx

```diff
@@ -202,6 +202,35 @@ export default class ReportHistoryStore {
         this.cache[reportID] = newCache.sort((a, b) => b.reportActionTimestamp - a.reportActionTimestamp);
     }
 
+    /**
+     * Merges history items using the reportActionID into the cache and creates it if it doesn't yet exist.
+     * We then sorts them by timestamp and then reprotActionID to ensure a consistent order if two report actions have the same timestamp
+     *
+     * @param {Number} reportID
+     * @param {Object[]} newHistory
+     */
+    mergeHistoryByReportActionID(reportID, newHistory) {
+        if (newHistory.length === 0) {
+            return;
+        }
+
+        const newCache = newHistory.reverse().reduce((prev, curr) => {
+            if (!prev.some((item) => item.reportActionID === curr.reportActionID)) {
```

Comment:
I don't get what the intention of this is. If the item did not exist, it adds it, but if it did exist, it does nothing and does not really update the cache?

Reply:
They do change, like when you edit a comment

Reply:
then I really don't get the need for all of this logic, why not just replace the cache with `newHistory`?